modules/nw-sriov-networknodepolicy-object.adoc

```diff
@@ -0,0 +1,105 @@
+
+<7> The number of the virtual functions (VF) to create for the SR-IOV physical network device. For an Intel Network Interface Card (NIC), the number of VFs cannot be larger than the total VFs supported by the device. For a Mellanox NIC, the number of VFs cannot be larger than `128`.
+
+<8> The `nicSelector` mapping selects the device for the Operator to configure. You do not need to specify values for all the parameters. It is recommended to identify the network device with enough precision to avoid selecting a device unintentionally.
+If you specify `rootDevices`, you must also specify a value for `vendor`, `deviceID`, or `pfNames`.
+If you specify both `pfNames` and `rootDevices` at the same time, ensure that they point to an identical device.
+
+<9> Optional: The vendor hex code of the SR-IOV network device. The only allowed values are either `8086` or `15b3`.
+
+<10> Optional: The device hex code of SR-IOV network device. The only allowed values are `158b`, `1015`, `1017`.
+
+<11> Optional: An array of one or more physical function (PF) names for the device.
+
+<12> An array of one or more PCI bus addresses for the PF of the device. Provide the address in the following format: `0000:02:00.1`.
+
+<13> Optional: The driver type for the virtual functions. You can specify one of the following values: `netdevice` or `vfio-pci`. The default value is `netdevice`.
++
+[NOTE]
+====
+For a Mellanox card to work in Data Plane Development Kit (DPDK) mode on bare metal nodes, use the `netdevice` driver type and set `isRdma` to `true`.
+====
+
+<14> Optional: Whether to enable remote direct memory access (RDMA) mode. The default value is `false`.
+
+<15> Optional: The link type for the VFs. You can specify one of the following values: `eth` or `ib`. `eth` is ethernet and `ib` is InfiniBand. The default value is `eth` if it is not explicitly set. When `linkType` is set to `ib`, `isRdma` will be set to `true` by SR-IOV Network Operator webhook automatically. When `linkType` is set to `ib`, `deviceType` should not be set to `vfio-pci`.
++
+[NOTE]
+====
+If `isRDMA` flag is set to `true`, you can continue to use the RDMA enabled VF as a normal network device.
+A device can be used in either mode.
+====
+
+[id="sr-iov-network-node-configuration-examples_{context}"]
+== SR-IOV network node configuration examples
+
+The following example describes the configuration for an IB device:
+
+.Example configuration for an IB device
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: policy-ib-net-1
+  namespace: openshift-sriov-network-operator
+spec:
+  resourceName: ibnic1
+  nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+  numVfs: 4
+  nicSelector:
+    vendor: "15b3"
+    deviceID: "101b"
+    rootDevices:
+      - "0000:19:00.0"
+  linkType: ib
+  isRdma: true
+----
```

Comment:
@jboxman do you think we shall have two examples, one for ib, the other for eth?
I checked 4.5 doc, there is no example.

Reply:
Looking at sriov network-attachment doc, we are putting examples in the nw-sriov-network-attachment, instead of having it in the object doc, shall we follow the same pattern and move the policy example in nw-sriov-configuring-device ?

Reply:
@zshi-redhat, yes, having an example for Ethernet would be great idea. Do you have any examples that I can include?

Reply:
@zshi-redhat, `modules/nw-sriov-network-attachment.adoc` includes conditionals so that it works for both IB and Ethernet. I wanted to include a limited example in the procedure, for context, but without any annotations for the object.

The only difference here is `kind: {rs}` is replaced with either `SriovNetwork` or `SriovIBNetwork`.

It looks like in `modules/nw-sriov-configuring-device.adoc` I don't do this. That's only because of expediency; docs freeze is on 13 October, and I haven't had time to add an example.